### PR TITLE
FIxing typos and refining naming again, getting ready to use a Payload

### PR DESCRIPTION
**Pull request by samcake**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a87d2927f17ccbe94d873c1342f86fadc52861b8 available at: https://gist.github.com/4eb3847aded6e54bb6ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4932.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4932.dmg

**Comment by samcake** [comment]
try it again

File: libraries/render/src/render/Scene.h [hunk 1]
```diff
@@ -120,58 +162,61 @@ class Item {
 // THe Payload class is the real Payload to be used
 // THis allow anything to be turned into a Payload as long as the required interface functions are available
 // When creating a new kind of payload from a new "stuff" class then you need to create specialized version for "stuff"
-// of th ePayload interface
-template <class T> const Item::Key payloadGetKey(const T* payload) { return Item::Key(); }
-template <class T> const Item::Bound payloadGetBound(const T* payload) { return Item::Bound(); }
-template <class T> void payloadRender(const T* payload) { }
+// of the Payload interface
+template <class T> const ItemKey payloadGetKey(const std::shared_ptr<T>& payloadData) { return ItemKey(); }
```
Comment (ZappoMan) [inline libraries/render/src/render/Scene.h]: maybe we should use QSharedPointer<>?

**Comment by samcake** [comment]
test this please

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: a87d2927f17ccbe94d873c1342f86fadc52861b8 available at: https://gist.github.com/4eb3847aded6e54bb6ce<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4932.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4932.dmg

**Pull request merged**